src/core/ext/transport/chttp2/transport/chttp2_transport.h

```diff
@@ -49,4 +49,13 @@ void grpc_chttp2_transport_start_reading(
     grpc_transport* transport, grpc_slice_buffer* read_buffer,
     grpc_closure* notify_on_receive_settings);
 
+namespace grpc_core {
+// Takes a bool input parameter which is true when a transport is being
+// constructed and false when a trasport is being destructed.
+typedef void (*TestOnlyGlobalHttp2TransportInitCallback)(bool);
```

Comment:
If you didn't want to go with the separate shutdown method, you could make this a bit more self-documenting by adding a parameter name, something like `is_constructing`

Reply:
Either way is fine with me